govcd/vm_test.go

```diff
@@ -967,15 +967,34 @@ func (vcd *TestVCD) Test_VMSetGetGuestCustomizationSection(check *C) {
 	guestCustomizationPropertyTester(vcd, check, vm)
 }
 
-// Test gathering VM virtual hardware items
-func (vcd *TestVCD) Test_GetVirtualHardwareSection(check *C) {
-	itemName := "TestGetVirtualHardwareSection"
+// Test create internal disk For VM
+func (vcd *TestVCD) Test_AddInternalDisk(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+
+	vm, storageProfile, diskSettings, diskId, err := vcd.createInternalDisk(check, 1)
+	check.Assert(err, IsNil)
+
+	//verify
+	disk, err := vm.GetInternalDiskById(diskId, true)
+	check.Assert(err, IsNil)
 
+	check.Assert(disk.StorageProfile.HREF, Equals, storageProfile.HREF)
+	check.Assert(disk.AdapterType, Equals, diskSettings.AdapterType)
+	check.Assert(*disk.ThinProvisioned, Equals, *diskSettings.ThinProvisioned)
+	check.Assert(*disk.Iops, Equals, *diskSettings.Iops)
+	check.Assert(disk.SizeMb, Equals, diskSettings.SizeMb)
+	check.Assert(disk.UnitNumber, Equals, diskSettings.UnitNumber)
+	check.Assert(disk.BusNumber, Equals, diskSettings.BusNumber)
+	check.Assert(disk.AdapterType, Equals, diskSettings.AdapterType)
+}
+
+func (vcd *TestVCD) createInternalDisk(check *C, busNumber int) (*VM, types.Reference, *types.DiskSettings, string, error) {
```

Comment:
Please add a comment with the function purpose and the description of the returned values

Reply:
added